src/Microsoft.DotNet.Build.Tasks.Packaging/src/HarvestPackage.cs

```diff
@@ -324,22 +324,48 @@ private bool ShouldSuppress(string packagePath)
         {
             if (_pathsToSuppress == null)
             {
-                _pathsToSuppress = PathsToSuppress.NullAsEmpty().Select(EnsureTrailingSlash).ToArray();
+                _pathsToSuppress = PathsToSuppress.NullAsEmpty().Select(EnsureDirectory).ToArray();
             }
 
             return _pathsToSuppress.Any(p => packagePath.StartsWith(p, StringComparison.OrdinalIgnoreCase));
         }
 
-        private static string EnsureTrailingSlash(string source)
+        private static string EnsureDirectory(string source)
         {
+            string result;
+
             if (source.Length < 1 || source[source.Length - 1] == '\\' || source[source.Length - 1] == '/')
             {
-                return source;
+                // already have a directory
+                result = source;
             }
             else
             {
-                return source + '/';
+                // could be a directory or file
+                var extension = Path.GetExtension(source);
+
+                if (extension != null && extension.Length > 0 && includedExtensions.Contains(extension, StringComparer.OrdinalIgnoreCase))
+                {
+                    // it's a file, find the directory portion
+                    var fileName = Path.GetFileName(source);
+                    if (fileName.Length != source.Length)
+                    {
+                        result = source.Substring(0, source.Length - fileName.Length);
+                    }
+                    else
+                    {
+                        // no directory portion, just return as-is
+                        result = source;
```

Comment:
It means it's a filename only, like "foo.dll".  We're only in this scope block if the string matched the extension pattern.